config_test.cc

```diff
@@ -0,0 +1,40 @@
+#include "gtest/gtest.h"
+#include "config.h"
+
+TEST(ServerConfigTest, SimpleConfig) {
+  ServerConfig config("test_config_1");
+  EXPECT_EQ(config.GetPortNo(), 12345);
+}
+
+TEST(ServerConfigTest, PortTooLargeConfig) {
+  try{
+  	ServerConfig config("test_config_2");
+  	EXPECT_EQ(config.GetPortNo(), 80);
+  }
+  catch (PortRangeException &e)
+  {
+	std::cout << e.what();
+  }
+}
+
+TEST(ServerConfigTest, NoPortConfig) {
+  try{
+  	ServerConfig config("test_config_3");
+  	EXPECT_EQ(config.GetPortNo(), 80);
+  }
+  catch (InvalidConfigException &e)
+  {
+	std::cout << e.what();
+  }
+}
+
+TEST(ServerConfigTest, EmptyConfig) {
+  try{
+  	ServerConfig config("test_config_4");
+  	EXPECT_EQ(config.GetPortNo(), 80);
+  }
+  catch (InvalidConfigException &e)
+  {
+	std::cout << e.what();
+  }
```

Comment:
Great unit tests! When we have to change this at all, the tests will help us determine that we don't break anything. Maybe try using a test fixture for future tests, as fixtures can make some unit tests easier to write, if they are similar.